rascal-lsp/src/main/rascal/util/LanguageServer.rsc

```diff
@@ -125,204 +109,132 @@ of the user.
 alias Focus = list[Tree];
 
 @synopsis{Function profile for outliner contributions to a language server}
+@deprecated{Only in use in deprecated functions.}
 alias Outliner         = list[DocumentSymbol] (Tree _input);
 
 @synopsis{Function profile for lenses contributions to a language server}
-@deprecated{The ((OrderedLensDetector)) has replaced this type}
+@deprecated{Only in use in deprecated functions.}
 alias LensDetector     = rel[loc src, Command lens] (Tree _input);
 
 @synopsis{Function profile for lenses contributions to a language server}
 alias OrderedLensDetector     = lrel[loc src, Command lens] (Tree _input);
 
 @synopsis{Function profile for executor contributions to a language server}
+@deprecated{Only in use in deprecated functions.}
 alias CommandExecutor  = value (Command _command);
 
 @synopsis{Function profile for inlay contributions to a language server}
+@deprecated{Only in use in deprecated functions.}
 alias InlayHinter      = list[InlayHint] (Tree _input);
 
-@synopsis{Function profile for documentation contributions to a language server}
-@description{
-A documenter is called on-demand, when documentation is requested by the IDE user.
-}
-@benefits{
-* is focused on a single documentation request, so does not need full program analysis.
-}
-@pitfalls{
-* should be extremely fast in order to provide interactive access.
-* careful use of `@memo` may help to cache dependencies, but this is tricky!
-}
-@deprecated{The ((FocusDocumenter)) has replaced this type.}
+@deprecated{Only in use in deprecated functions}
 alias Documenter = set[str] (loc _origin, Tree _fullTree, Tree _lexicalAtCursor);
 
-@synopsis{Function profile for documentation contributions to a language server}
```

Comment:
is this great documentation now lost? is there no place to keep these around?  

(same for all the other details that was present in the aliases below)

Reply:
Some of it has been moved to the LanguageServer documentation where it belongs. That can be extended more; I'm thinking of a table with a row per service.